File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/Poison.lua
function GetName()
  return "Poison"
end

function GetType()
  return Debuff
end

function GetHookPoints()
  return { TurnTriggered }
end

function OnAttachment(char)
end

function OnRemoval(char) -- Can only be removed by Cleansing Effects
end

function Trigger(hookPoint, char)
  if hookPoint ~= TurnTriggered then
    return
  end

  char:TakeTrueDamage(char:GetCurrentHP() * 0.2)

  if char:GetCurrentHP() <= 0 then
    char:Heal(1)
  end

end

File: src/ProjectR.Scripting/content/scripts/afflictions/Debuffs/Poison.cs
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class Poison : Affliction
    {
        public override string Name { get { return "Poison"; } }

        public override AfflictionType Type { get { return AfflictionType.Debuff; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TurnTriggered, 
                };
            }
        }

        protected override void OnTurnTriggered(ICharacter character)
        {
            character.TakeTrueDamage(character.CurrentHP * 0.25);

            if (character.CurrentHP <= 0)
            {
                character.Heal(1);
            }
        }
    }
}